### DISCUSSION TITLE
WiX v3 - Upgrade does not fail even if base setup uninstallation fails

### DISCUSSION THREAD

**Discussion by yuvnith** [wixtoolset/issues#8212] [Questions]
We have two setups: base and upgrade. During the upgrade, we ran into a situation where the uninstallation of the base setup failed, but the upgrade went through without any issues. Base and upgrade version become the two entries in the control panel as a result. 

The log file says 
```
Applying execute package: {bfbdf7c4-fb65-42e4-b6f3-0841b9851353}, action: Uninstall, path: C:\ProgramData\Package Cache\{bfbdf7c4-fb65-42e4-b6f3-0841b9851353}\SetupRTM.exe, arguments: '"C:\ProgramData\Package Cache\{bfbdf7c4-fb65-42e4-b6f3-0841b9851353}\SetupRTM.exe" -uninstall -quiet -burn.related.upgrade -burn.ancestors={6917f766-6515-4ff6-94cf-83560aaefc64} -burn.filehandle.self=1032'
[A85C:C8C8][2024-05-16T10:31:30]e000: Error 0xffffffff: Process returned error: 0xffffffff
[A85C:C8C8][2024-05-16T10:31:30]e000: Error 0xffffffff: Failed to execute EXE package.
[EF1C:8A78][2024-05-16T10:31:30]e000: Error 0xffffffff: Failed to configure per-machine EXE package.
[EF1C:8A78][2024-05-16T10:31:30]w350: Applied non-vital package: {bfbdf7c4-fb65-42e4-b6f3-0841b9851353}, encountered error: 0xffffffff. Continuing...
[A85C:C8C8][2024-05-16T10:31:30]i372: Session end, registration key: SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{6917f766-6515-4ff6-94cf-83560aaefc64}, resume: ARP, restart: None, disable resume: No
[A85C:C8C8][2024-05-16T10:31:30]i371: Updating session, registration key: SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\{6917f766-6515-4ff6-94cf-83560aaefc64}, resume: ARP, restart initiated: No, disable resume: No
```

How can I get the upgrade setup to fail when the base setup uninstall fails? Could someone please assist me with this?

**Comment by ssrmm** [comment]
Your log says `Applied non-vital package: [...]`, which probably means that you defined the package with `Vital="no"`. If so, that seems consistent with the documented behavior of the attribute:
> Vital ([YesNoTypeUnion](https://wixtoolset.org/docs/schema/wxs/yesnotype/)) : Specifies whether the package must succeed for the chain to continue. The default "yes" indicates that if the package fails then the chain will fail and roll back or stop. If "no" is specified then the chain will continue even if the package reports failure.
> &mdash; https://wixtoolset.org/docs/schema/wxs/exepackage/

To change that you'd have to set `Vital="yes"` or leave it unspecified (since that's the default). If the base setup is already released, then you can't really change that anymore unfortunately, at least not for upgrades from that released version.

**Reply by yuvnith** [reply]
Hi @ssrmm  Vital="yes" is applicable for packages which we configure in our bundle. But the non-vital package here is my base setup and it is a bundle. I think we cannot set this base setup as a vital package to my upgrade setup 

Please correct me if I am wrong
